HashSet.java

```diff
@@ -0,0 +1,45 @@
+// Time Complexity :   add(), remove(), contains()  ->  O(1)
+// Space Complexity :  O(n) I am using HashMap to simulate the behavior
+// Did this code successfully run on Leetcode : Yes
+// Any problem you faced while coding this : None
+
+
+// Your code here along with comments explaining your approach
+class MyHashSet {
+    
+    HashMap<Integer, Integer> index;        // Using HashMap to simulate the Hashset behavior
```

Comment:
Whenever we are asked to make hashing data structure. The intention of interviewer is to find if you can take a basic data structure and implement hashing on it. 

We shouldn't be implementing hashset based on a hashmap

Reply:
Does an ArrayList implementation work in this case?

Reply:
Why thinking about ArrayList, (still a higher level DS). We will be making it work in O(1) using a normal array

Reply:
Oh okay. 
So, an ArrayList implementation also, if it checks with contains() function on whether the key exists or not and then perform 'add'/ HashSet put() or 'return the key' / HashSet get(), both the HashSet operations will have O(n) implementation resulting the HashSet to have O(n) implementation as well. Is this Correct?

Reply:
I didn't understand. Please ask question in class